File: Problem-1/Solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}

        for i, currentNum in enumerate(nums):
            if currentNum in hashMap:
                return [hashMap[currentNum], i]
            else:
                hashMap[target - currentNum] = i
        return []

if __name__ == "__main__":
    try_sol = Solution()
    answer = try_sol.twoSum([3, 10, 13, -3], 0)
    print(answer)

File: Problem-1/Solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut my_map = HashMap::new();

        let len = nums.len();
        let mut current_num = 0;

        for i in 0..len {
            current_num = nums[i];

            match my_map.get(&current_num) {
                Some(&value) => return vec![value, i as i32],
                None => {
                    my_map.insert(target - current_num, i as i32);
                }
            }
        }
        
        Vec::new()
    }
}

fn main() {
    
}

